addons/spreadsheet_account/static/src/accounting_functions.js

```diff
@@ -298,3 +315,36 @@ functionRegistry.add("ODOO.ACCOUNT.GROUP", {
         return accountTypes.join(",");
     },
 });
+
+functionRegistry.add("ODOO.RESIDUAL", {
+    description: _t("Return the residual amount for the specified account(s) and period"),
+    args: ODOO_RESIDUAL_ARGS(),
+    category: "Odoo",
+    returns: ["NUMBER"],
+    compute: function (
+        accountCodes,
+        dateRange,
+        offset = { value: 0 },
+        companyId = { value: null },
+        includeUnposted = { value: false }
+    ) {
+        const _accountCodes = toString(accountCodes)
+            .split(",")
+            .map((code) => code.trim())
+            .sort();
+        const _offset = toNumber(offset, this.locale);
+        const _dateRange = parseAccountingDate(dateRange?.value, this.locale);
+        const _companyId = companyId?.value;
```

Comment:
`const _companyId = toNumber(companyId, this.locale);` is better and safer. The user might have input the string `"1"` in the function